pkg/templatelookup/availableModules.go

```diff
@@ -0,0 +1,76 @@
+package templatelookup
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+)
+
+var (
+	ErrInvalidModuleInSpec   = errors.New("invalid configuration in Kyma spec.modules")
+	ErrInvalidModuleInStatus = errors.New("invalid module entry in Kyma status")
+)
+
+type AvailableModule struct {
+	v1beta2.Module
+	Enabled         bool
+	ValidationError error
+}
+
+func (a AvailableModule) IsInstalledByVersion() bool {
+	return a.configuredWithVersionInSpec() || a.installedwithVersionInStatus()
+}
+
+// configuredWithVersionInSpec returns true if the Module is enabled in Spec using a specific version instead of a channel.
+func (a AvailableModule) configuredWithVersionInSpec() bool {
+	return a.Enabled && a.Version != "" && a.Channel == ""
+}
+
+// installedwithVersionInStatus returns true if the Module installed using a specific version (instead of a channel) is reported in Status.
+func (a AvailableModule) installedwithVersionInStatus() bool {
+	return !a.Enabled && shared.NoneChannel.Equals(a.Channel) && a.Version != ""
+}
+
+func GetAvailableModules(kyma *v1beta2.Kyma) []AvailableModule {
```

Comment:
If you refactor this `GetAvailableModules` not belong to kyma struct, I think it's better to rename it, not use `GetXXX`, it's not aligned with the naming conversion.

Reply:
done.